src/k8s-engine/lima.ts

```diff
@@ -1554,7 +1554,44 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
         ]);
         if (config.enabled) {
           this.lastCommandComment = 'Checking k3s images';
-          await this.progressTracker.action(this.lastCommandComment, 100, this.k3sHelper.ensureK3sImages(desiredVersion));
+          try {
+            await this.progressTracker.action(this.lastCommandComment, 100, this.k3sHelper.ensureK3sImages(desiredVersion));
+          } catch (ex:any) {
+            console.log(`Failed to find version ${ desiredVersion.raw }: ${ ex }`, ex);
+            if (ex.code === 'ECONNREFUSED' || ex.toString().includes('getaddrinfo ENOTFOUND github.com') || ex.toString().includes('getaddrinfo EAI_AGAIN github.com')) {
+              try {
+                const newVersion: semver.SemVer = await this.k3sHelper.selectClosestImage(desiredVersion);
```

Comment:
This should be in the `desiredVersion` getter, instead of here.

Reply:
Right now the `desiredVersion` getter uses the pref and the list of available versions to select something. But we don't know that the desired version isn't actually available until we try to ensure it exists in `ensureK8sImages`. So I think this is ok for now.

Reply:
Hmm. Can we then change it to have `ensureK8sImages` return the resulting version, and go from there, instead? Doing a try/catch here seems rather ugly (and more importantly, this method is already pretty giant).